build/BuildSteps.cs

```diff
@@ -341,7 +341,7 @@ public static void Test()
 
             Environment.SetEnvironmentVariable("DURABLE_FUNCTION_PATH", Settings.DurableFolder);
 
-            Shell.Run("dotnet", $"test {Settings.TestProjectFile} -f net8.0 --logger trx");
+            Shell.Run("dotnet", $"test {Settings.TestProjectFile} -f net8.0 --logger trx --filter \"{TestingTraits.TraitName.Category}!={TestingTraits.TestCategory.FinalIntegrationTests}\"");
```

Comment:
Note that this only fixes this for this specific build tool and will have problems elsewhere, such as when someone uses Test Explorer or does a `dotnet run` themselves.

I would instead recommend adjusting the test project default filters appropriately to exclude the trait, leaving the existing line here, and then adding a line to have a filter just to the new trait. You can see an example of this in the Worker - For SDK container builds, I had to adjust the configuration to avoid introducing a docker daemon dependency in normal test paths.

Reply:
Here's the PR for the worker changeset I was referring to: https://github.com/Azure/azure-functions-dotnet-worker/pull/2671/files

Reply:
thanks! I made the according changes